src/common/addlayers/ServerService.js

```diff
@@ -668,11 +668,15 @@ var SERVER_SERVICE_USE_PROXY = true;
     this.addSearchResultsForHyper = function(server, filterOptions) {
       var searchUrl = service_.catalogList[0].url + '_search?';
       if (filterOptions !== null) {
-        searchUrl = service_.catalogList[filterOptions.catalogKey].url + '_search?';
+        var catalogKey = filterOptions.catalogKey;
+        if (!isNaN(Number(catalogKey))) {
```

Comment:
sorry, this is not the way to fix the failing test.
The failure is the expected outcome. We need to fix the test not check if the catalogKey exist. It does always exist and this needs to blow up if it does not!